src/lib/workspace/ComputationRoute.tsx

```diff
@@ -142,20 +191,36 @@ export function ComputationRoute(props: Props) {
                   if (analysisState.analysis == null) return;
                   const computations =
                     analysisState.analysis.descriptor.computations;
-                  const computation = createComputation(
-                    app.name,
-                    name,
-                    configuration,
-                    computations
+                  /* 
+                    Here we're checking for a matching configuration before instantiating a new one
+                    If there's a match, we'll store it as existingConfig and use that data in the else block
+                  */
+                  // @ts-ignore
+                  const existingConfig = computations.find((c) =>
+                    isEqual(c.descriptor.configuration, configuration)
```

Comment:
Whenever we're looking to see if an app instance already exists, we should also check to verify it has the same app type. I noticed that if i first make an XY-Relationships app instance (configuration=null) and then try to make a Counts and Proportions app instance (also configuration=null) i'm sent back to the original XY-Relationships app instance.